maven-plugin/src/main/java/org/l2x6/cq/maven/CreateExampleMojo.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2020 CQ Maven Plugin
+ * project contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.l2x6.cq.maven;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+import java.util.stream.Stream;
+import org.apache.maven.model.Model;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.l2x6.cq.common.CqCommonUtils;
+
+/**
+ * Scaffolds a new Camel Quarkus example project.
+ *
+ * @since 4.7.2
+ */
+@Mojo(name = "create-example", requiresProject = false)
+public class CreateExampleMojo extends AbstractMojo {
+    static final String DEFAULT_TEMPLATES_URI_BASE = "classpath:/create-example-templates";
+
+    /**
+     * Directory where the new example should be added. Default is the current directory of the running Java process.
+     *
+     * @since 4.7.2
+     */
+    @Parameter(property = "cq.basedir", defaultValue = "${project.basedir}")
+    File basedir;
+    private Path basePath;
+
+    /**
+     * The default encoding used to process example project templates.
+     *
+     * @since 4.7.2
+     */
+    @Parameter(defaultValue = CqUtils.DEFAULT_ENCODING, required = true, property = "cq.encoding")
+    String encoding;
+    Charset charset;
+
+    /**
+     * The base directory where example project templates are located. Can be used to override the plugin default template
+     * content.
+     *
+     * @since 4.7.2
```

Comment:
I'd normally bump the minor with a new mojo. But I would not mind keeping this as is if you do not want to bother with changing the `@since`.

Reply:
No worries - I have bumped the version to 4.8.0.